### PR TITLE
Postgres: $all, $and CLP and more

### PR DESCRIPTION
**Pull request by flovilmart**

### CODE REVIEW

**Comment by facebook-github-bot-1** [comment]
@flovilmart updated the pull request - [view changes](https://github.com/ParsePlatform/parse-server/pull/2551/files/d6c2154e0043eb46fc134a2451a6e0c9be57f89b..f6ed47cf9bf3980c30d9c7b85ed3a2cdeb09a927)

**Comment by facebook-github-bot-6** [comment]
@flovilmart updated the pull request - [view changes](https://github.com/ParsePlatform/parse-server/pull/2551/files/f6ed47cf9bf3980c30d9c7b85ed3a2cdeb09a927..f69d9b47978ddd42578a36a96e48bc3f670280f5)

**Comment by facebook-github-bot-5** [comment]
@flovilmart updated the pull request - [view changes](https://github.com/ParsePlatform/parse-server/pull/2551/files/f69d9b47978ddd42578a36a96e48bc3f670280f5..d4a76702fabc66bb6b6efac1e76e7982f1f9ecbc)

**Comment by facebook-github-bot-3** [comment]
@flovilmart updated the pull request - [view changes](https://github.com/ParsePlatform/parse-server/pull/2551/files/d4a76702fabc66bb6b6efac1e76e7982f1f9ecbc..b6b58508b924d8048ae2d499fcd70d9f5ae272cb)

**Comment by facebook-github-bot-3** [comment]
@flovilmart updated the pull request - [view changes](https://github.com/ParsePlatform/parse-server/pull/2551/files/b6b58508b924d8048ae2d499fcd70d9f5ae272cb..066ac3575123ae9122c928385ce073023fe73a94)

**Comment by facebook-github-bot-8** [comment]
@flovilmart updated the pull request - [view changes](https://github.com/ParsePlatform/parse-server/pull/2551/files/066ac3575123ae9122c928385ce073023fe73a94..c43c73550b80e66cb310fd65f06b31e51e8ee2e6)

**Comment by facebook-github-bot-5** [comment]
@flovilmart updated the pull request - [view changes](https://github.com/ParsePlatform/parse-server/pull/2551/files/c43c73550b80e66cb310fd65f06b31e51e8ee2e6..6cc581d0ea450c9b52e8813b69f7065f6ebae735)

**Comment by facebook-github-bot-8** [comment]
@flovilmart updated the pull request - [view changes](https://github.com/ParsePlatform/parse-server/pull/2551/files/6cc581d0ea450c9b52e8813b69f7065f6ebae735..458a2220eab9524000bca6082fbedc0d0ef88681)

**Comment by facebook-github-bot-9** [comment]
@flovilmart updated the pull request - [view changes](https://github.com/ParsePlatform/parse-server/pull/2551/files/458a2220eab9524000bca6082fbedc0d0ef88681..6cc581d0ea450c9b52e8813b69f7065f6ebae735)

File: spec/ParseAPI.spec.js [hunk 1]
```diff
@@ -1496,6 +1495,7 @@ it_exclude_dbs(['postgres'])('ensure that if you try to sign up a user with a un
     });
   });
 
+  // TODO: do we want that on PG as well?
```
Comment (drew-gross) [inline spec/ParseAPI.spec.js]: Probably not, it's basically a legacy data format

Comment (flovilmart) [inline spec/ParseAPI.spec.js]: I'll remove the line note and wrap it

**Comment by facebook-github-bot-7** [comment]
@flovilmart updated the pull request - [view changes](https://github.com/ParsePlatform/parse-server/pull/2551/files/6cc581d0ea450c9b52e8813b69f7065f6ebae735..dd621aa103fb7dba57ada8165acef607c1b91a3a)

**Comment by facebook-github-bot-5** [comment]
@flovilmart updated the pull request - [view changes](https://github.com/ParsePlatform/parse-server/pull/2551/files/dd621aa103fb7dba57ada8165acef607c1b91a3a..3e6a39a7ec4d8afe01b780b260f4982fe09a2fe2)

**Pull request merged**